Source/messaging/Logging.h

```diff
@@ -47,7 +47,7 @@ namespace Logging {
     do {                                                                                                                                     \
         using __control__ = SYSLOG_CONTROL(CATEGORY);                                                                                        \
         if (__control__::IsEnabled() == true) {                                                                                              \
-            static_assert(std::is_base_of<WPEFramework::Logging::BaseCategory, CATEGORY>::value, "SYSLOG() only for Logging controls");      \
+            static_assert(CATEGORY::Type == WPEFramework::Core::Messaging::MessageType::LOGGING, "SYSLOG() only for Logging controls");      \
```

Comment:
This will still break the build if one uses SYSLOG() with a custom category not inheriting from BaseCategoryType, but with a bit misleading error (i.e. no Type present), hence the original is_base_of. This is more than enough, but I think is_base_of would still work with the BaseCategory being a template.